platform/darwin/src/MGLRendererConfiguration.mm

```diff
@@ -0,0 +1,119 @@
+    return [self localFontFamilyNameWithInfoDictionaryObject:infoDictionaryObject];
+}
+
+- (nullable NSString *)localFontFamilyNameWithInfoDictionaryObject:(nullable id)infoDictionaryObject {
+    if ([infoDictionaryObject isKindOfClass:[NSNumber class]] && ![infoDictionaryObject boolValue]) {
+        // NO means don’t use local fonts.
+        return nil;
+    } else if ([infoDictionaryObject isKindOfClass:[NSString class]]) {
+        return infoDictionaryObject;
+    } else if ([infoDictionaryObject isKindOfClass:[NSArray class]]) {
+        // mbgl::LocalGlyphRasterizer::Impl accepts only a single string, but form a cascade list with one font on each line.
+        return [infoDictionaryObject componentsJoinedByString:@"\n"];
+    }
+    
+#if TARGET_OS_IPHONE
+    return [UIFont systemFontOfSize:0 weight:UIFontWeightRegular].familyName;
+#else
+    return [NSFont systemFontOfSize:0 weight:NSFontWeightRegular].familyName;
+#endif
+}
+
+- (BOOL)perSourceCollisions {
+    id infoDictionaryObject = [NSBundle.mainBundle objectForInfoDictionaryKey:MGLCollisionBehaviorPre4_0Key];
+    return [self perSourceCollisionsWithInfoDictionaryObject:infoDictionaryObject];
+}
+
+- (BOOL)perSourceCollisionsWithInfoDictionaryObject:(nullable id)infoDictionaryObject {
+    // Set the collision behaviour. A value set in `NSUserDefaults.standardUserDefaults`
+    // should override anything in the application's info.plist
+    if ([NSUserDefaults.standardUserDefaults objectForKey:MGLCollisionBehaviorPre4_0Key]) {
+        return [NSUserDefaults.standardUserDefaults boolForKey:MGLCollisionBehaviorPre4_0Key];
+    } else if ([infoDictionaryObject isKindOfClass:[NSNumber class]] || [infoDictionaryObject isKindOfClass:[NSString class]]) {
+        // Also support NSString to correspond with the behavior of `-[NSUserDefaults boolForKey:]`
+        return [infoDictionaryObject boolValue];
+    }
+    return NO;
+}
+
+- (MGLGlyphsRasterizationMode)glyphsRasterizationMode {
+    id infoDictionaryObject = [NSBundle.mainBundle objectForInfoDictionaryKey:MGLGlyphsRasterizationModeKey];
+    return [self glyphsRasterizationModeWithInfoDictionaryObject:infoDictionaryObject];
+}
+
+- (MGLGlyphsRasterizationMode)glyphsRasterizationModeWithInfoDictionaryObject:(id)infoDictionaryObject {
+    if (!infoDictionaryObject || ![infoDictionaryObject isKindOfClass:[NSString class]]) {
+        return MGLGlyphsRasterizationModeNone;
+    }
+    NSDictionary *nameOptionMap = @{@"MGLNoGlyphsRasterizedLocally":@(MGLGlyphsRasterizationModeIdeographsRasterizedLocally),
+                                    @"MGLIdeographsRasterizedLocally":@(MGLGlyphsRasterizationModeNoGlyphsRasterizedLocally),
+                                    @"MGLAllGlyphsRasterizedLocally":@(MGLGlyphsRasterizationModeAllGlyphsRasterizedLocally)};
+
+    return (MGLGlyphsRasterizationMode)[nameOptionMap[infoDictionaryObject] integerValue];
```

Comment:
For other readers: this should result in `MGLGlyphsRasterizationModeNone` if `infoDictionaryObject` is not expected.

See: https://github.com/mapbox/mapbox-gl-native-ios/pull/517/files#diff-944ba4f88de8974791fb7cf7e890a87157c51513eb6e31179983619a8312c6a6R176